tests/test_logging.py

```diff
@@ -611,3 +616,86 @@ async def test_print_default_logging_cmd(inmanta_config, tmp_path):
         normal_config_stdout = stdout.decode("utf-8")
         # Assert that the outputs are equal
         assert normal_config_stdout == tty_config_stdout
+
+
+@pytest.fixture
+def setup_compiler_logging(tmpdir, monkeypatch):
+    """
+    Set up the logging config for the compiler to use a specific formatter.
+    """
+    compiler_logging_config = """
+        disable_existing_loggers: false
+        formatters:
+          console_formatter:
+            format: "COMPILER_CONFIG_FLAG -- %(message)s"
+        handlers:
+          console_handler:
+            class: logging.StreamHandler
+            formatter: console_formatter
+            level: DEBUG
+            stream: ext://sys.stdout
+        root:
+          handlers:
+          - console_handler
+          level: INFO
+        version: 1
+    """
+    compiler_logging_config_file = os.path.join(tmpdir, "compiler.yml")
+    with open(compiler_logging_config_file, "w") as fh:
+        fh.write(compiler_logging_config)
+    config = os.path.join(tmpdir, "logging_config.yml")
+    with open(config, "w") as fh:
+        fh.write(
+            f"""
+[logging]
+compiler = {os.path.abspath(compiler_logging_config_file)}
+        """
+        )
+    Config.load_config(min_c_config_file=config)
```

Comment:
Bit hackish to load the config twice ? Once here and once in the `inmanta_config` fixture ? Is there a better alternative ?

Reply:
I think this question is no longer relevant, because with the above-mentioned issues resolved, I don't think the test case will still work. The config will be overridden by the start of the server.

Reply:
Maybe you could overwrite the `server_config` fixture to make it load the right config.

Reply:
Overwrote the `inmanta_config` fixture.

Reply:
I had to move the test to its own file, I couldn't get it to play nice with the other tests calling into the `inmanta_config` fixture.